providers/factories/oracle/websocket.go

```diff
@@ -0,0 +1,118 @@
+package oracle
+
+import (
+	"fmt"
+	"math/big"
+	"net/http"
+
+	"github.com/skip-mev/slinky/oracle/config"
+	"github.com/skip-mev/slinky/pkg/math"
+	apihandlers "github.com/skip-mev/slinky/providers/base/api/handlers"
+	wshandlers "github.com/skip-mev/slinky/providers/base/websocket/handlers"
+	wsmetrics "github.com/skip-mev/slinky/providers/base/websocket/metrics"
+	"github.com/skip-mev/slinky/providers/types/factory"
+	"github.com/skip-mev/slinky/providers/websockets/bitfinex"
+	"github.com/skip-mev/slinky/providers/websockets/bitstamp"
+	"github.com/skip-mev/slinky/providers/websockets/bybit"
+	coinbasews "github.com/skip-mev/slinky/providers/websockets/coinbase"
+	"github.com/skip-mev/slinky/providers/websockets/cryptodotcom"
+	"github.com/skip-mev/slinky/providers/websockets/gate"
+	"github.com/skip-mev/slinky/providers/websockets/huobi"
+	"github.com/skip-mev/slinky/providers/websockets/kraken"
+	"github.com/skip-mev/slinky/providers/websockets/kucoin"
+	"github.com/skip-mev/slinky/providers/websockets/mexc"
+	"github.com/skip-mev/slinky/providers/websockets/okx"
+	oracletypes "github.com/skip-mev/slinky/x/oracle/types"
+	"go.uber.org/zap"
+)
+
+func OracleWebSocketQueryHandlerFactory() factory.WebSocketQueryHandlerFactory[oracletypes.CurrencyPair, *big.Int] {
+	return func(logger *zap.Logger, cfg config.ProviderConfig, wsMetrics wsmetrics.WebSocketMetrics) (wshandlers.WebSocketQueryHandler[oracletypes.CurrencyPair, *big.Int], error) {
```

Comment:
This is all copy pasted from the previous implementation but now moved into a separate function.